scripts/check-go-version-yaml.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# Function to check if the YAML file contains the specified Go version after
+# field 'go:'.
+check_go_version_yaml() {
+    local yamlfile="$1"
+    local required_go_version="$2"
+
+    # Use grep to find lines with 'go:'. The grep exist status is ignored.
+    local go_lines=$(grep -i '^\s*go:\s*"[0-9]\+\.[0-9]\+\.[0-9]\+"' "$yamlfile" || true)
+
+    # Check if any lines specify the Go version.
+    if [ -n "$go_lines" ]; then
+        # Extract the Go version from the file's lines. Example matching strings:
+        # go: "1.21.0"
+        local extracted_go_version=$(echo "$go_lines" | grep -oP '^\s*go:\s*"\K[^"]+')
+
+        # Check if the extracted Go version matches the required version.
+        if [ "$extracted_go_version" != "$required_go_version" ]; then
+            echo "Error: $yamlfile specifies Go version '$extracted_go_version', but not version '$required_go_version'."
+            exit 1
+        else
+            echo "$yamlfile specifies Go version $required_go_version."
+        fi
+    fi
+}
+
+# Function to check if the YAML file contains the specified Go version after
+# environment variable 'GO_VERSION:'.
+check_go_version_env_variable() {
+    local yamlfile="$1"
+    local required_go_version="$2"
+
+    # Use grep to find lines with 'GO_VERSION:'. The grep exist status is
+    # ignored.
+    local go_lines=$(grep -i 'GO_VERSION:' "$yamlfile" || true)
+
+    # Check if any lines specify the Go version.
+    if [ -n "$go_lines" ]; then
+        # Extract the Go version from the file's lines. Example matching strings:
+        # GO_VERSION: "1.21.0"
+        # GO_VERSION: '1.21.0'
+        # GO_VERSION: 1.21.0
+        # GO_VERSION:1.21.0
+        #   GO_VERSION:1.21.0
```

Comment:
nit: remove the spaces

Reply:
the leading spaces here are intentional. We need to be robust to leading white space.